alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderSelectCustomFunctionAction.java

```diff
@@ -27,35 +27,39 @@
 import com.synopsys.integration.alert.common.descriptor.DescriptorMap;
 import com.synopsys.integration.alert.common.descriptor.config.field.LabelValueSelectOption;
-import com.synopsys.integration.alert.common.descriptor.config.field.endpoint.SelectCustomEndpoint;
 import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
 import com.synopsys.integration.alert.common.enumeration.DescriptorType;
-import com.synopsys.integration.alert.common.exception.AlertException;
+import com.synopsys.integration.alert.common.rest.HttpServletContentWrapper;
 import com.synopsys.integration.alert.common.rest.model.FieldModel;
+import com.synopsys.integration.alert.common.security.authorization.AuthorizationManager;
+import com.synopsys.integration.exception.IntegrationException;
 
 @Component
-public class ProviderSelectCustomEndpoint extends SelectCustomEndpoint {
+public class ProviderSelectCustomFunctionAction extends CustomFunctionAction<List<LabelValueSelectOption>> {
     private DescriptorMap descriptorMap;
 
     @Autowired
-    public ProviderSelectCustomEndpoint(CustomEndpointManager customEndpointManager, DescriptorMap descriptorMap) throws AlertException {
-        super(ChannelDistributionUIConfig.KEY_PROVIDER_NAME, customEndpointManager);
+    public ProviderSelectCustomFunctionAction(AuthorizationManager authorizationManager, DescriptorMap descriptorMap) {
+        super(authorizationManager);
         this.descriptorMap = descriptorMap;
     }
 
     @Override
-    protected List<LabelValueSelectOption> createData(FieldModel fieldModel) throws AlertException {
-        return descriptorMap.getDescriptorByType(DescriptorType.PROVIDER).stream()
-                   .map(descriptor -> descriptor.createMetaData(ConfigContextEnum.DISTRIBUTION))
-                   .flatMap(Optional::stream)
-                   .map(descriptorMetadata -> new LabelValueSelectOption(descriptorMetadata.getLabel(), descriptorMetadata.getName()))
-                   .sorted()
-                   .collect(Collectors.toList());
+    public ActionResponse<List<LabelValueSelectOption>> createActionResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) throws IntegrationException {
+        List<LabelValueSelectOption> content = descriptorMap.getDescriptorByType(DescriptorType.PROVIDER).stream()
+                                                   .map(descriptor -> descriptor.createMetaData(ConfigContextEnum.DISTRIBUTION))
+                                                   .flatMap(Optional::stream)
+                                                   .map(descriptorMetadata -> new LabelValueSelectOption(descriptorMetadata.getLabel(), descriptorMetadata.getName()))
+                                                   .sorted()
+                                                   .collect(Collectors.toList());
+        return new ActionResponse<>(HttpStatus.OK, content);
```

Comment:
For the sake of consistency, it might be nice to wrap the `List<LabelValueSelectOption>` in an object to make content consistent.